### PR TITLE
fix(#92,#93,#203): validate channel names, improve empty-state copy, and add agent guide

### LINKED ISSUE #92: test: add backend validation test for invalid channel names
## Context

Channel creation accepts a name parameter, but edge cases (empty string, only spaces, special characters, extremely long names) may not be fully validated or tested.

## Expected change

Add one or more integration tests in backend/tests/ that verify the API rejects invalid channel names with appropriate error responses.

## Suggested files

- backend/tests/channels_test.rs (or create a new test file)
- backend/src/services/channel.rs (to inspect current validation logic)

## Acceptance criteria

- [ ] Test for empty channel name returns 400
- [ ] Test for channel name exceeding max length returns 400
- [ ] Test for channel name with invalid characters returns 400
- [ ] Tests run with cargo test --lib and pass

### LINKED ISSUE #93: frontend: improve empty-state text for first workspace/channel
## Context

When a user first logs in and no channels exist yet, the empty-state message in the sidebar or main canvas may be generic or unhelpful. A clearer message helps new users understand what to do next.

## Expected change

Improve the empty-state copy in the channel list and/or main content area when no channels are present.

## Suggested files

- frontend/src/features/channels/ (or similar feature directory)
- Look for empty-state or placeholder components

## Acceptance criteria

- [ ] Empty state explains that no channels exist yet
- [ ] Empty state suggests creating a channel or joining an existing one
- [ ] Copy is friendly and matches the project's tone
- [ ] Visual appearance remains consistent with the rest of the UI

### LINKED ISSUE #203: Harden repository for agent-guided contribution workflow
## Context

The repository structure and `AGENTS.md` are now in place. The next step is to make RustChat consistently usable by human contributors and AI coding agents such as Codex, Claude Code, or similar tools.

This should not add unnecessary ceremony. The goal is a lean, practical contribution workflow inspired by mature Rust projects such as Turso: clear repo map, clear testing expectations, predictable CI, and explicit agent guardrails.

## Scope

- Review `AGENTS.md` against the current repository layout and make sure all paths, crate names, and commands are accurate.
- Ensure `README.md` and `CONTRIBUTING.md` point contributors to `AGENTS.md` where appropriate.
- Add or update lightweight agent/contributor docs, for example:
  - `docs/agent-guides/testing.md`
  - `docs/agent-guides/code-quality.md`
  - `docs/agent-guides/pr-workflow.md`
  - `docs/architecture/overview.md`
- Document the expected local validation commands:
  - `cargo fmt --check`
  - `cargo clippy --workspace --all-features --all-targets -- --deny warnings`
  - `cargo test --workspace`
- Review CI and make sure the same basic quality gates are enforced.
- Add or update PR / issue templates if missing.

## RustChat-specific notes

Please make the contribution guidance explicit about the main RustChat boundaries:

- realtime/chat server behavior
- API and protocol changes
- auth/session handling
- persistence/storage behavior
- frontend/client-facing behavior, if applicable

Protocol or storage changes should include migration notes or compatibility notes where relevant.

## Acceptance criteria

- [ ] `AGENTS.md` accurately reflects the current RustChat repo layout.
- [ ] The main build, lint, format, and test commands are documented and work locally.
- [ ] CI enforces the same baseline quality gates.
- [ ] `README.md` / `CONTRIBUTING.md` link to the agent/contributor workflow.
- [ ] At least one short agent guide exists for testing and PR workflow.
- [ ] The repo makes it clear what an AI agent may change and what requires human review.
- [ ] The guidance stays lean and does not introduce unnecessary process overhead.

### PR DESCRIPTION
**Pull request by zoorpha**
Closes #92, closes #93, closes #203.

**Backend (#92)**
- Adds `validate_channel_name()` in `backend/src/api/v4/channels/crud.rs` and calls it from `create_channel`.
- Rejects empty/whitespace-only names, names longer than 64 characters, and names containing characters other than lowercase letters, numbers, hyphens and underscores.
- Trims whitespace-padded valid names before persisting; adds a test that `"  trimmed-channel  "` returns 200 and is stored as `"trimmed-channel"`.
- Adds integration test `create_channel_rejects_invalid_names` in `backend/tests/api_v4_channels_all.rs` covering empty, whitespace-only, over-long and invalid-character names (all 400), plus a valid-name regression guard (200).

**Frontend (#93)**
- Replaces the sidebar "No channels" placeholder with friendly, actionable copy: "No channels yet" / "Create your first channel or browse existing ones to get started."

**Contributor / Agent Workflow (#203)**
- Creates canonical root `AGENTS.md` with repo map, CI-aligned validation commands, and AI-agent change boundaries.
- Links `README.md` and `CONTRIBUTING.md` to `AGENTS.md`.
- Adds `docs/agent-guides/README.md` as a thin index to existing guides.
- Deprecates `docs/internal/AGENTS.md` in favor of the root file.

**Verification**
- `cargo test --test api_v4_channels_all` passes.
- `cargo fmt --all -- --check`, `cargo clippy --all-targets --all-features -- -D warnings`, and `cargo test --lib` pass.
- `npm run docs:check-links`, `npm run docs:check-config`, and `npm run docs:build` pass.
- Frontend build/unit tests were not run in this environment due to Node version constraints.

### CODE REVIEW

**Review by senolcolak — DISMISSED** [review]

**Review by senolcolak — DISMISSED** [review]

**Review by senolcolak — DISMISSED** [review]

**Comment by senolcolak** [comment]
@codex review

File: backend/src/api/v4/channels/crud.rs [hunk 1]
```diff
@@ -144,6 +144,34 @@ fn default_channel_type() -> String {
     "O".to_string()
 }
 
+/// Validates a channel name. Channel names must be non-empty, at most 64
+/// characters, and contain only lowercase letters, digits, hyphens and
+/// underscores. This matches Mattermost URL-name conventions and keeps us
+/// safely under the 255-character database limit.
+fn validate_channel_name(name: &str) -> Result<(), AppError> {
+    let trimmed = name.trim();
```
[the remark this answers is not in the document]
Comment (zoorpha) [reply backend/src/api/v4/channels/crud.rs:152]: Fixed in `b967a5e5`: `create_channel` now trims the name before validation and passes the normalized value to `ChannelRepository::create`. Added an integration-test assertion that whitespace-padded valid names are accepted and stored trimmed.

File: docs/agent-guides/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Agent Guides
+
+This directory is the landing point for AI-agent-specific guidance in RustChat.
+
+For the canonical contributor and agent entry point, see the root [`AGENTS.md`](https://github.com/kubedoio/rustchat/blob/main/AGENTS.md).
```
[the remark this answers is not in the document]
Comment (zoorpha) [reply docs/agent-guides/README.md:5]: Intentional: root `AGENTS.md` lives outside the Vitepress `docs/` site root, so a relative `../../AGENTS.md` link is reported as a dead link during `npm run docs:build`. Absolute URLs to the current repository (`kubedoio/rustchat`) are required here to keep the docs build passing.

**Review by senolcolak — APPROVED** [review]
